app/routes/auth/contract.js

```diff
@@ -1,4 +1,45 @@
 import Ember from 'ember'
 
+/**
+ * Status Code References
+ * 510 - The user does not have a github account (Reason = 5)
+ * 520 - The user has already signed up (Reason = 4)
+ */
+
+const USER_DOES_NOT_EXIST = 5
+const ALREADY_HAS_ACCOUNT = 4
+const ERROR_ROUTE = 0
+
 export default Ember.Route.extend({
+  queryParams: {
+    username: {
+      refreshModel: true
+    }
+  },
+  model: function (params) {
+    return this.get('store').queryRecord('developer',
+      {
+        username: params.username,
+        token: '!'
+      })
+      .then((res) => {
+        return {
+          username: params.username
+        }
+      })
+      .catch((err) => {
+        if (err.errors[0].status === '510') {
```

Comment:
These custom Error-codes are a Bad Idea. The situation you have on your hands is a `403`, which semantically is `unauthorized`. You should simply return a `403`. Extra details (e.g. 'user does not exist' or 'user already has an account') should be provided by the server and not by the client.